dcicutils/obfuscation_utils.py

```diff
@@ -96,25 +154,30 @@ def obfuscate_dict(dictionary: dict, inplace: bool = False, show: bool = False,
     check_true(not obfuscated or is_obfuscated(obfuscated),
                message=f"If obfuscated= is supplied, it must be {OBFUSCATED_VALUE_DESCRIPTION}.")
 
-    def has_values_to_obfuscate(dictionary: dict) -> bool:
-        for key, value in dictionary.items():
-            if isinstance(value, dict):
-                if has_values_to_obfuscate(value):
-                    return True
-            elif isinstance(value, str) and should_obfuscate(key) and not is_obfuscated(value):
-                return True
-        return False
+    if show:
+        return item
+
+    orig_item = item
+
+    changed = StorageCell(False)
+
+    if not inplace:
+        item = copy.deepcopy(item)
+
+    def process_recursively(item):
+        # We only need to process non-atomic items recursively, since they are the only things
+        # that might conceivably be or contain a dictionary in need of obfuscation.
+        if isinstance(item, dict):
+            for key, value in item.items():
+                if should_obfuscate(key):
```

Comment:
I think we might want this to avoid obfuscating values which appear already to be obfuscated: ```if should_obfuscate(key) and not is_obfuscated(value):```

Reply:
And actually, technically, since we can pass in a specific `obfuscated` value to use, the `is_obfuscated` function should also check if the given value is equal to this given `obfuscated` value, no?

Reply:
> `if should_obfuscate(key) and not is_obfuscated(value):`

Again, there were no unit tests covering this case. But isn't this a job for `should_obfuscate`?